spec/models/defect_spec.rb

```diff
@@ -320,4 +320,17 @@ class TestDefect < Defect
       end
     end
   end
+
+  describe '::TRADES' do
+    it 'includes all the trades, and only those trades, that are in the trades-by-category lists' do
```

Comment:
This is a good test as it covers all the contents rather than sampling. It would have caught the problem initially.

The implementation is rigid by choosing to sort both lists to check for an absolute match. This has pros and cons. It gives us high confidence but may end up being breaking in future when/if ordering changes deliberately.

This shouldn't hold up the merge but I would consider the use of `include` where the ordering is not the part under test. Something like:

```
expect(Defect::TRADES).to include(Defect::ELECTRICAL_TRADES)
expect(Defect::TRADES).to include(Defect::PLUMBING_TRADES)
expect(Defect::TRADES).to include(Defect::CARPENTRY_TRADES)
expect(Defect::TRADES).to include(Defect::COSMETIC_TRADES
```